logger.js

```diff
@@ -79,7 +79,7 @@ logger.dump_and_exit = function (code) {
 
 logger.log = function (level, data) {
     if (level === 'PROTOCOL') {
-        data = data.replace(/\n/g, '\\n\n');
+        data = data.replace(/\n/g, '\\n');
```

Comment:
(btw, these are questions of curiosity. You've given more thought to this than anyone else in a long while so I'm inclined to +1 this PR on that basis alone.)